src/main/kotlin/me/avo/realworld/kotlin/ktor/data/ArticleBasic.kt

```diff
@@ -0,0 +1,14 @@
+package me.avo.realworld.kotlin.ktor.data
+
+data class ArticleBasic(val title: String , val body: String , val description: String) {
+
+    fun CompareAndUpdate(article : ArticleDetails): ArticleDetails {
+        var newArt : ArticleDetails = article
```

Comment:
I think this should be merged into one assignment. 
Use copy once and put the conditions inline.